aztec/src/main/kotlin/org/wordpress/aztec/watchers/event/sequence/ObservationQueue.kt

```diff
@@ -0,0 +1,65 @@
+package org.wordpress.aztec.watchers.event.sequence
+
+import android.os.Build
+import org.wordpress.aztec.watchers.event.IEventInjector
+import org.wordpress.aztec.watchers.event.buckets.API26Bucket
+import org.wordpress.aztec.watchers.event.text.TextWatcherEvent
+import org.wordpress.aztec.watchers.event.buckets.Bucket
+
+class ObservationQueue(injector: IEventInjector) : EventSequence<TextWatcherEvent>() {
+    val buckets = ArrayList<Bucket>()
+    val injector = injector
+
+    init {
+        init()
+    }
+
+    override fun add(element: TextWatcherEvent): Boolean {
+        synchronized(this@ObservationQueue) {
+            val added: Boolean = super.add(element)
+            if (added) {
+                processQueue()
+            }
+            return added
+        }
+    }
+
+    private fun processQueue() {
+        // here let's check whether our current queue matches / fits any of the installed buckets
+        var foundOnePartialMatch = false
```

Comment:
Worth to add a check on the size of the `buckets` here, and return immediately when = 0?
```
        if (buckets.size == 0)
            return
```

Reply:
I think that's what already happens in `for (bucket in buckets) { ...` when `buckets` is of size == 0; it doesn't go through the loop